Apsitvarkom.DataAccess/Migrations/20221218115321_DatabaseTriggers.cs

```diff
@@ -0,0 +1,342 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace Apsitvarkom.DataAccess.Migrations
+{
+    /// <inheritdoc />
+    public partial class DatabaseTriggers : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql(
+                "create function Update_PollutedLocationAlreadyCleanedUp() " +
+                "returns trigger as " +
+                "$$ begin " +
+                   "if ( " +
+                   "select \"PollutedLocations\".\"Progress\" = 100 from \"PollutedLocations\" " +
+                   "where \"PollutedLocations\".\"Id\" = new.\"Id\" " +
+                   ") " +
+                   "then " +
+                   "raise exception 'Could not update the polluted location, as its cleaning up process is already complete (Progress = 100).'; " +
+                   "end if; " +
+                   "return new; " +
+                "end; $$ " +
+                "language plpgsql; " +
+
+                "create trigger Update_PollutedLocationAlreadyCleanedUp " +
+                "before update on \"PollutedLocations\" " +
+                "for each row " +
+                "execute procedure Update_PollutedLocationAlreadyCleanedUp();"
+            );
```

Comment:
Why not multiline string instead of concat?

https://stackoverflow.com/a/1100265/13135665